camelot/core/files/storage.py

```diff
@@ -51,24 +51,25 @@ class StoredFile:
     its methods should block.
     """
 
-    def __init__(self, storage: 'Storage', name: str):
+    def __init__(self, storage: 'Storage', name: PurePosixPath):
         """
         :param storage: the storage in which the file is stored
         :param name: the key by which the file is known in the storage
         """
+        assert isinstance(name, PurePosixPath)
         self.storage = storage
-        self.name = name
+        self.name: PurePosixPath = name
 
     @property
     def verbose_name(self) -> str:
         """The name of the file, as it is to be displayed in the GUI"""
-        return self.name
+        return self.name.as_posix()
 
     def __getstate__(self) -> Dict[str, str]:
         """Returns the key of the file. To support pickling stored files
         in the database in a :class:`camelot.model.memento.Memento` object
         """
-        return {'name': self.name}
+        return {'name': self.verbose_name}
```

Comment:
dit moet hier self.name blijven ... verbose name is UI stuff, mss wel nt onverstandig om de name hier om te zetten naar een tuple of een 'string' of zo